cmd/agent-mobile/pkg/api/storage.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package api
+
+// StoreIterator is the iterator for the latest snapshot of the underlying store.
+type StoreIterator interface {
+	// Next moves the iterator to the next key/value pair.
+	// It returns false if the iterator is exhausted.
+	Next() bool
+
+	// Free releases associated resources. Release should always success
+	// and can be called multiple times without causing error.
+	Free()
+
+	// Error returns any accumulated error. Exhausting all the key/value pairs
+	// is not considered to be an error.
+	Error() error
+
+	// Key returns the key of the current key/value pair, or nil if done.
+	// The caller should not modify the contents of the returned slice, and
+	// its contents may change on the next call to any 'seeks method'.
+	Key() []byte
+
+	// Value returns the value of the current key/value pair, or nil if done.
+	// The caller should not modify the contents of the returned slice, and
+	// its contents may change on the next call to any 'seeks method'.
+	Value() []byte
+}
+
+// Store is the storage interface.
+type Store interface {
+	// Put stores the key and the record
+	Put(k string, v []byte) error
+
+	// Get fetches the record based on key
+	Get(k string) ([]byte, error)
+
+	// Iterator returns an iterator for the latest snapshot of the
+	// underlying store
+	//
+	// Args:
+	//
+	// startKey: Start of the key range, include in the range.
+	// endKey: End of the key range, not include in the range.
+	//
+	// Returns:
+	//
+	// StoreIterator: iterator for result range
+	Iterator(startKey, endKey string) StoreIterator
+
+	// Delete will delete a record with k key
+	Delete(k string) error
+}
+
+// StorageProvider defines storage interface.
+type StorageProvider interface {
+	// OpenStore opens a store with given name space and returns the handle
+	OpenStore(name string) (Store, error)
+
+	// CloseStore closes store of given name space
+	CloseStore(name string) error
+
+	// Close closes all stores created under this store provider
+	Close() error
+}
```

Comment:
Can't we use Storage APIs defined in [aries](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/storage/store.go#L24-L83) rather than defining again ?

Reply:
No, we can't. This is why I implemented wrapper. See https://github.com/trustbloc/agent-sdk/pull/49/files#diff-e6a0bb16b03023e283d0651d80a675f44a7c581ac6da7ca557b9a7965d3d91acR20

Reply:
Oh ok. As I understand from our chat, the reason for this is that "Release method is not allowed for iOS".

Reply:
Yeah, this is certainly the reason. But also we cannot use interfaces from different packages. I am not 100% sure. But next time I will tell you more.